tests/sim/main.rs

```diff
@@ -68,6 +75,63 @@ fn init() -> (
     (account, empty_account_1, empty_account_2, request_proxy)
 }
 
+// Initialize test environment with 3 accounts (alice, bob, builder), a fungible conversion mock, and a fungible token mock.
+fn init_fungible() -> (
+    UserAccount,
+    UserAccount,
+    UserAccount,
+    ContractAccount<FungibleConversionProxyContract>,
+    ContractAccount<FungibleTokenContractContract>,
+) {
+    let genesis = GenesisConfig::default();
+    let root = init_simulator(Some(genesis));
+
+    deploy!(
+        contract: FPOContractContract,
+        contract_id: "mockedfpo".to_string(),
+        bytes: &MOCKED_BYTES,
+        signer_account: root,
+        deposit: to_yocto("7")
+    );
+
+    let ft_contract = deploy!(
+        contract: FungibleTokenContractContract,
+        contract_id: "mockedft".to_string(),
+        bytes: &MOCKED_BYTES,
+        signer_account: root,
+        deposit: to_yocto("7")
+    );
+
+    let account = root.create_user("alice".to_string(), to_yocto(DEFAULT_BALANCE));
+    let empty_account_1 = root.create_user("bob".parse().unwrap(), to_yocto(DEFAULT_BALANCE));
+    let empty_account_2 = root.create_user("builder".parse().unwrap(), to_yocto(DEFAULT_BALANCE));
+
+    let fungible_request_proxy = deploy!(
+        contract: FungibleConversionProxyContract,
+        contract_id: FUNGIBLE_CONTRACT_ID,
+        bytes: &FUNGIBLE_REQUEST_PROXY_BYTES,
+        signer_account: root,
+        deposit: to_yocto("5"),
```

Comment:
@intrprtr I'm concerned about what you said in a previous comment:

> Since these two calls only use 1 yoctoNEAR each per call, a simple intial deposit of some negligible value worth of NEAR into the contract will be enough to ensure it functions properly forever, so we don't have to worry about what is or isn't attached by the calling fungible token contract.

I wish there was a test where the fungibleConversionProxy has 0 yocto and alice pays the 2 yocto, just to show that it's possible (in case the fungibleConversionProxy's NEAR reserves run out).

More generally, I wish there were tests where the entry point was `ft_transfer` instead of `ft_on_transfer`